src/components/AddAssignment.js

```diff
@@ -1,125 +1,102 @@
-        super(props);
-        this.state = { assignments: [] };
-      };
+      super(props);
+      this.state = {courseId: '', assignmentName: '', dueDate: '' };
+      this.handleChange = this.handleChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    };
+  
+
+    //gets values from form
+    handleChange(event) {
+      const target = event.target;
+      const value = target.value;
+      const name = target.name;
+      this.setState({
+        [name]: value
+      });
+    }
 
-    // when submit button pressed, send new assignment to back end 
-    handleSubmit = ( ) => {
-        console.log("AddAssignment.handleSubmit");
-        const token = Cookies.get('XSRF-TOKEN');
-        
-        fetch(`http://localhost:8081//assignment/createAssignment?AssignmentName=${this.state.AssignmentName}&AssignmentDueDate=${this.state.AssignmentDueDate}&CourseId=${this.state.AssignmentCourseID}` , 
-            {  
-              method: 'POST', 
-              headers: { 'Content-Type': 'application/json',
-                         'X-XSRF-TOKEN': token }, 
-              body: JSON.stringify({assignmentId:this.props.location.assignment.assignmentId,  assignments: this.state.assignments})
-            } )
-        .then(res => {
-            if (res.ok) {
-              toast.success("Assignment successfully Created", {
-              position: toast.POSITION.BOTTOM_LEFT
-              });
-            
-            } })
-          .catch(err => {
-            toast.error("Failed to Create Assignment", {
-              position: toast.POSITION.BOTTOM_LEFT
-            });
-            console.error(err);
-          });
-     };        
 
-    // when user has entered a new grade, update the state
-    //  id    - index of row of grade change
-    //  props - contains the new grade
-    handleEditCellChange = ({ id, field, props }) => {
-        console.log("edit cell change id:"+id+" field:"+field+" value:"+props.value);
-        const newgrades = this.state.grades.map(r => {
-          if (r.id === id){
-            return {...r, [field]:props.value};
+ 
+  handleSubmit(event){
+      event.preventDefault();
+      console.log("Adding New Assignment");
+      console.log(JSON.stringify({courseId:this.state.courseId, assignmentName:this.state.assignmentName, dueDate:this.state.dueDate}));
+      const token = Cookies.get('XSRF-TOKEN');
+      
+      //putting params in URL because I used Get Requests in backend method 
+      fetch(`http://localhost:8081//assignment/createAssignment?AssignmentName=${this.state.assignmentName} 3&AssignmentDueDate=${this.state.dueDate}&CourseId=${this.state.courseId}` , 
+          {  
+            method: 'POST', 
+            headers: { 'Content-Type': 'application/json',
+                       'X-XSRF-TOKEN': token }, 
+            body: JSON.stringify({courseId:this.state.courseId, assignmentName:this.state.assignmentName, dueDate:this.state.dueDate})
+          } )
```

Comment:
do you need to send the JSON in the body too? It seems like you are already sending the data via the url parameters?

Reply:
otherwise this looks good to me and you can merge!